tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,12 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (NonEmptySet (xs:Set<int>)) =
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
@ploeh no that isn't right. It should produce a uniform distribution across all possible permutations of the input, including itself. In other words, `[0;1]` or `[1;0]` must be equally likely.

I think we just check 1) that it is a permutation and 2) that if we take a list of length 5 and produce  10 permutations (i.e. sample 10), that at least one of those permutations is different from any of the others.

The chance of this *not* holding in case of a uniform shuffle - in other words, the chance of producing 10 times the same input list of length 5 - is (1/5!)^10, which is small enough. (unless my probability is off, which it may very well be :) )